src/components/dataTable.js

```diff
@@ -469,15 +469,22 @@
     };
 
     const renderTableHead = () => {
-      if ((loading && !loadOnScroll) || error) {
+      if (loading && !loadOnScroll) {
         return Array.from(Array(children.length).keys()).map((colIdx) => (
           <TableCell key={colIdx}>
-            <div className={classes.skeleton}>
-              {error && displayError && error.message}
-            </div>
+            <div className={classes.skeleton} />
           </TableCell>
         ));
       }
+
+      if (error) {
+        return (
+          <TableCell colSpan={children.length}>
```

Comment:
you are missing a key prop here like how its done above.
`<TableCell key={colIdx}>`

Reply:
That is only needed for mapped components from a array @M-Ryan92